src/eckit/exception/Exceptions.h

```diff
@@ -253,6 +253,14 @@ class UnexpectedState : public Exception {
     explicit UnexpectedState(const std::string&, const CodeLocation& = {});
 };
 
+/// Signals a protocol error in client/server communiation.
+/// Use if the expected communication protocol is broken, i.e. wrong message flow or broken message.
+/// @note This exception is 'silent', i.e. no backtrace is logged.
+class ProtocolError : public Exception {
+public:
+    explicit ProtocolError(const std::string& what);
```

Comment:
Please accept a location here. All exceptions thrown should (ideally) include the location.

Reply:
> Please accept a location here. All exceptions thrown should (ideally) include the location.

Hah! that's because I've made that exact change on the previous PR ;-)